### PR TITLE
Incremental 02b - Unit tests for kokkos_free

### PR DESCRIPTION
**Pull request by rgayatri23**

### CODE REVIEW

**Comment by dalg24** [comment]
OK to test

File: core/unit_test/Makefile [hunk 1]
```diff
@@ -306,6 +306,9 @@ endif
 
 ifeq ($(KOKKOS_INTERNAL_USE_SERIAL), 1)
     OBJ_SERIAL = UnitTestMainInit.o gtest-all.o
+    OBJ_SERIAL += TestSerial_Incremental_01_ExecutionSpace.o
```
Comment (jeffmiles63) [inline core/unit_test/Makefile]: not needed for incremental tests

**Comment by jeffmiles63** [comment]
see comments in #2536

File: core/unit_test/incremental_testing/Test02b_MemorySpace.hpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  const int N = 100000;
+  const int M = 100000;
+
+  void testit_free() {
+    for(int i = 0; i < N; ++i)
+    {
+      double *data = (double*) Kokkos::kokkos_malloc<MemSpace>("data", M*sizeof(double));
+      ASSERT_FALSE(data == nullptr);
+      Kokkos::kokkos_free<MemSpace>(data);
+    }
```
Comment (dalg24) [inline core/unit_test/incremental_testing/Test02b_MemorySpace.hpp]: Please add a comment about the intent here. I assume you allocate many times enough memory such that the device would run out of memory if de-allocation does not happen. The cumulated amount of memory allocated is 80GB (assuming double are 8 bytes).

File: core/unit_test/incremental_testing/Test02b_MemorySpace.hpp [hunk 3]
```diff
@@ -47,21 +47,22 @@
 #include <type_traits>
 #include <gtest/gtest.h>
 
-
 namespace Test {
 
-// Test construction and assignment
+// Unit test for kokkos_free
+// We constantly allocate and de-allocate memory.
+// If the kokkos_free does not free the memory, we will exceed the available
```
Comment (mhoemmen) [inline core/unit_test/incremental_testing/Test02b_MemorySpace.hpp]: This is only true if failing to deallocate would result in allocating more than the entire virtual memory space.  Linux overcommit means that if you don't actually touch the allocations, then you can allocate way more than the physical memory capacity.

**Comment by rgayatri23** [comment]
All the requested changes have been addressed.

**Comment by rgayatri23** [comment]
Retest this please.

File: cmake/kokkos_test_cxx_std.cmake [hunk 4]
```diff
@@ -35,9 +35,18 @@ FUNCTION(kokkos_set_cxx_standard_feature standard)
     GLOBAL_SET(KOKKOS_CXX_STANDARD_FEATURE "")
   ENDIF()
 
-  IF(NOT ${FEATURE_NAME} IN_LIST CMAKE_CXX_COMPILE_FEATURES)
-    MESSAGE(FATAL_ERROR "Compiler ${KOKKOS_CXX_COMPILER_ID} should support ${FEATURE_NAME}, but CMake reports feature not supported")
-  ENDIF()
+    IF(NOT ${FEATURE_NAME} IN_LIST CMAKE_CXX_COMPILE_FEATURES)
+          IF (KOKKOS_CXX_COMPILER_ID STREQUAL "NVIDIA")
+                  MESSAGE(STATUS "nvcc_wrapper does not support TARGET_COMPILE_FEATURES")
+                        GLOBAL_SET(KOKKOS_CXX_STANDARD_FEATURE "")
```
Comment (jjwilke) [inline cmake/kokkos_test_cxx_std.cmake]: I don't understand how this branch got brought back in. nvcc_wrapper should work now. Were there issues happening that required this change?

Comment (dalg24) [reply cmake/kokkos_test_cxx_std.cmake]: `kokkos:incremental-testing` branch should not be touching that file at all.  We will need to do some history cleaning...

Comment (masterleinad) [reply cmake/kokkos_test_cxx_std.cmake]: A diff between https://github.com/kokkos/kokkos/pull/2633 and `develop` shows that the current changes are pretty self-contained and reasonably small. I guess we just rewrite the history for `incremental-testing` completely before merging into `develop`. At least, rebasing is a nightmare.

Comment (crtrott) [reply cmake/kokkos_test_cxx_std.cmake]: I revert all those changes in #2633

**Pull request merged**